### PR TITLE
Clean up usage of 'String', 'String.Equals', and 'String.Compare'

### PR DESCRIPTION
**Pull request by paulvanbrenk**

### CODE REVIEW

File: Nodejs/Product/Nodejs/Jade/Text/TextStream.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public override string ToString()
         public string GetText(int position, int length)
         {
             if (length == 0)
-                return String.Empty;
```
Comment (jramsay) [inline Nodejs/Product/Nodejs/Jade/Text/TextStream.cs]: add braces

File: Common/Product/TestAdapter/VisualStudioApp.cs [hunk 2]
```diff
@@ -119,7 +119,7 @@ private static DTE GetDTE(int processId)
                         // Do nothing, there is something in the ROT that we do not have access to.
                     }
 
-                    if (!string.IsNullOrEmpty(name) && string.Equals(name, progId, StringComparison.Ordinal))
+                    if (!StringComparer.Ordinal.Equals(name, progId))
```
Comment (jramsay) [inline Common/Product/TestAdapter/VisualStudioApp.cs]: The condition looks different - now you no longer have a null check and it is looking for !equals

File: Common/Product/SharedProject/MsBuildProjectElement.cs [hunk 3]
```diff
@@ -183,14 +183,7 @@ public override bool Equals(object obj)
             var include1 = GetMetadata(ProjectFileConstants.Include);
             var include2 = msBuildProjElem.GetMetadata(ProjectFileConstants.Include);
 
-            // Unfortunately the checking for nulls have to be done again, since neither String.Equals nor String.Compare can handle nulls.
```
Comment (jramsay) [inline Common/Product/SharedProject/MsBuildProjectElement.cs]: aren't the null checks still required? Equals will return true if they are both null which was not the case previously

File: Common/Product/SharedProject/MsBuildProjectElement.cs [hunk 4]
```diff
@@ -70,10 +70,10 @@ protected override string ItemType
         /// <param name="attributeValue">Value to give to the attribute</param>
         public override void SetMetadata(string attributeName, string attributeValue)
         {
-            Debug.Assert(String.Compare(attributeName, ProjectFileConstants.Include, StringComparison.OrdinalIgnoreCase) != 0, "Use rename as this won't work");
+            Debug.Assert(StringComparer.OrdinalIgnoreCase.Equals(attributeName, ProjectFileConstants.Include), "Use rename as this won't work");
```
Comment (jramsay) [inline Common/Product/SharedProject/MsBuildProjectElement.cs]: missing !

File: Common/Product/SharedProject/CommonProjectNode.cs [hunk 5]
```diff
@@ -636,11 +636,11 @@ private void UpdateShowAllFiles(HierarchyNode node, bool enabled)
         {
             bool? showAllFiles = null;
             var showAllFilesSetting = showAllFilesValue.Trim();
-            if (String.Equals(showAllFilesSetting, CommonConstants.ProjectFiles))
+            if (StringComparer.OrdinalIgnoreCase.Equals(showAllFilesSetting, CommonConstants.ProjectFiles))
```
Comment (jramsay) [inline Common/Product/SharedProject/CommonProjectNode.cs]: Equals by default is:
ordinal (case-sensitive and culture-insensitive) comparison.
You're now ignoring case here and below - is this deliberate?

File: Common/Product/SharedProject/CommonFileNode.cs [hunk 6]
```diff
@@ -42,10 +42,14 @@ public bool IsFormSubType
             get
             {
                 var result = this.ItemNode.GetMetadata(ProjectFileConstants.SubType);
-                if (!String.IsNullOrEmpty(result) && string.Compare(result, ProjectFileAttributeValue.Form, true, CultureInfo.InvariantCulture) == 0)
+                if (!string.IsNullOrEmpty(result) && StringComparer.InvariantCultureIgnoreCase.Equals(result, ProjectFileAttributeValue.Form))
```
Comment (jramsay) [inline Common/Product/SharedProject/CommonFileNode.cs]: IgnoreCase - is that a deliberate change?

File: Common/Product/SharedProject/CommonEditorFactory.cs [hunk 7]
```diff
@@ -243,7 +243,7 @@ private IntPtr CreateDocumentView(string documentMoniker, string physicalView, I
                 // create code window as default physical view
                 return CreateCodeView(documentMoniker, textLines, ref editorCaption, ref cmdUI);
             }
-            else if (string.Compare(physicalView, "design", true, CultureInfo.InvariantCulture) == 0)
+            else if (StringComparer.InvariantCultureIgnoreCase.Equals(physicalView, "design"))
```
Comment (jramsay) [inline Common/Product/SharedProject/CommonEditorFactory.cs]: IgnoreCase - is that a deliberate change?

File: Common/Product/SharedProject/CommonProjectNode.cs [hunk 8]
```diff
@@ -636,11 +636,11 @@ private void UpdateShowAllFiles(HierarchyNode node, bool enabled)
...
             {
                 showAllFiles = false;
             }
-            else if (String.Equals(showAllFilesSetting, CommonConstants.ShowAllFiles))
+            else if (StringComparer.OrdinalIgnoreCase.Equals(showAllFilesSetting, CommonConstants.ShowAllFiles))
```
Comment (jramsay) [inline Common/Product/SharedProject/CommonProjectNode.cs]: same comment here

File: Common/Product/SharedProject/ProjectNode.cs [hunk 9]
```diff
@@ -1717,20 +1717,11 @@ public virtual void Load(string fileName, string location, string name, uint fla
                         // We have to be sure that we are not going to lose data here. If the project name is a.b.c then for a project that was based on a zipped template(the wizard calls us) GetFileNameWithoutExtension will suppress "c".
                         // We are going to check if the parameter "name" is extension based and the extension is the same as the one from the "filename" parameter.
                         var tempExtension = Path.GetExtension(name);
-                        if (!string.IsNullOrEmpty(tempExtension))
```
Comment (jramsay) [inline Common/Product/SharedProject/ProjectNode.cs]: this is now different if they're both null - intentional?

File: Common/Product/SharedProject/ProjectNode.cs [hunk 10]
```diff
@@ -3449,7 +3440,7 @@ internal int CompareNodes(HierarchyNode node1, HierarchyNode node2)
 
             if (node1.SortPriority == node2.SortPriority)
             {
-                return string.Compare(node2.Caption, node1.Caption, true, CultureInfo.CurrentCulture);
+                return StringComparer.CurrentCultureIgnoreCase.Compare(node2.Caption, node1.Caption);
```
Comment (jramsay) [inline Common/Product/SharedProject/ProjectNode.cs]: IgnoreCase wasn't there before - intentional?

Comment (paulvanbrenk) [inline Common/Product/SharedProject/ProjectNode.cs]: `true` means ignore case in the original

File: Common/Product/SharedProject/CommonFileNode.cs (hunk 6 above)
Comment (paulvanbrenk) [inline Common/Product/SharedProject/CommonFileNode.cs]: `true` in original means ignore case

File: Common/Product/SharedProject/CommonEditorFactory.cs (hunk 7 above)
Comment (paulvanbrenk) [inline Common/Product/SharedProject/CommonEditorFactory.cs]: `true` in original means ignore case

File: Common/Product/TestAdapter/VisualStudioApp.cs [hunk 11]
```diff
@@ -119,7 +118,7 @@ private static DTE GetDTE(int processId)
                         // Do nothing, there is something in the ROT that we do not have access to.
                     }
 
-                    if (!string.IsNullOrEmpty(name) && string.Equals(name, progId, StringComparison.Ordinal))
+                    if (!string.IsNullOrEmpty(name) && !StringComparer.Ordinal.Equals(name, progId))
```
Comment (jramsay) [inline Common/Product/TestAdapter/VisualStudioApp.cs]: I think this is still off, no?  It shouldn't have the !

Comment (jramsay) [inline Common/Product/TestAdapter/VisualStudioApp.cs]: I think this is still off, no?  It shouldn't have the ! for StringComparer.Ordinal.Equals(name, progId))

Comment (paulvanbrenk) [inline Common/Product/TestAdapter/VisualStudioApp.cs]: progId is never null, so the null check is no longer needed.

File: Common/Product/TestAdapter/VisualStudioApp.cs [hunk 12]
```diff
@@ -119,7 +118,7 @@ private static DTE GetDTE(int processId)
                         // Do nothing, there is something in the ROT that we do not have access to.
                     }
 
-                    if (!string.IsNullOrEmpty(name) && string.Equals(name, progId, StringComparison.Ordinal))
+                    if (StringComparer.Ordinal.Equals(name, progId))
```
Comment (jramsay) [inline Common/Product/TestAdapter/VisualStudioApp.cs]: Looks slightly off now - should be:
if (!string.IsNullOrEmpty(name) && StringComparer.Ordinal.Equals(name, progId))

**Pull request merged**